components/ILIAS/Chatroom/classes/class.ilChatroomAppEventListener.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Chatroom\CleanUpBans;
+
+class ilChatroomAppEventListener implements ilAppEventListener
+{
+    private static ?ilChatroomAppEventListener $instance = null;
+
+    private readonly CleanUpBans $clean_up;
+
+    public function __construct(ilDBInterface $database)
+    {
+        $this->clean_up = new CleanUpBans($this->dbCachePrepare($database));
+    }
+
+    /**
+     * @param array<string,mixed>  $a_parameter
+     */
+    public static function handleEvent(string $component, string $event, array $parameter): void
+    {
+        global $DIC;
+        self::$instance ??= new self($DIC->database());
+        self::$instance->handle($component, $event, $parameter);
+    }
+
+    public function handle(string $component, string $event, array $parameter): void
```

Comment:
```suggestion
    private function handle(string $component, string $event, array $parameter): void
```

Reply:
Hm, I don't think that this method should be private.
`handle` is the entry point into the code, even though it is only used in `handleEvent`.
Do you want this method to be private because it isn't used anywhere else? Or because the main entry point into the event handler should be `handleEvent`? IMO this forces to use the Database from the `$DIC`.

If `handle` should be private then `__construct` should also be private, because a class instance would be useless anyway.

Reply:
IMO `handleEvent` should be the only `public` method. You are right, `__construct` should be private as well.